### PR TITLE
Za - remember-login-details

### PR DESCRIPTION
**Pull request by Akintobi41**
# Why this PR?
This PR implements the "Remember Me" functionality for the login form with security. Login details ( password) are encrypted before being saved to localStorage. When a user returns to the login page, the encrypted password is decrypted and used to populate the login fields.

# What does this PR do?
This PR implements the "Remember Me" functionality for the login form with encrypted data storage. 

# How to test your work?
-Verify that login details are encrypted and saved correctly when "Remember Me" is selected.
-Confirm that login fields are pre-filled with decrypted data on revisiting the page.

# Pages to see your changes?
/auth/login

### CODE REVIEW

File: src/components/pages/Auth/Login/hooks-login/useRememberMe.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+type TRememberMeActions = {
+  rememberMe: boolean;
+  setTag: Dispatch<SetStateAction<string>>;
+  setPassword: Dispatch<SetStateAction<string>>;
+};
+
+type TRememberMeProps = TUserCredentials & TRememberMeActions;
+
+const key = import.meta.env.VITE_ENCRYPT_KEY;
```
Comment (ibraheemosule) [inline src/components/pages/Auth/Login/hooks-login/useRememberMe.ts:17]: rename the env to CRYPTOJS_KEY

File: src/components/pages/Auth/Login/hooks-login/useRememberMe.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+function saveUser({ tag, password }: TUserCredentials) {
+  const prevUser = localStorage.getItem('skoolizy-USER');
```
Comment (ibraheemosule) [inline src/components/pages/Auth/Login/hooks-login/useRememberMe.ts:20]: The user doesn't need to be in caps. You can't just say skoolizy_user

File: src/components/pages/Auth/Login/hooks-login/useRememberMe.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  if (prevUser) localStorage.clear();
+
+  if (!tag || !password) return;
+  localStorage.setItem(
+    'skoolizy-USER',
```
Comment (ibraheemosule) [inline src/components/pages/Auth/Login/hooks-login/useRememberMe.ts:26]: ditto

File: src/components/pages/Auth/Login/hooks-login/useRememberMe.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    JSON.stringify({
+      tag,
+      password: CryptoJS.AES.encrypt(password, key).toString(),
```
Comment (ibraheemosule) [inline src/components/pages/Auth/Login/hooks-login/useRememberMe.ts:29]: why are you setting toString()

File: src/components/pages/Auth/Login/hooks-login/useRememberMe.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    })
+  );
+}
+
+function isUserSaved(): { tag: string; password: string } {
+  const data = localStorage.getItem('skoolizy-USER');
```
Comment (ibraheemosule) [inline src/components/pages/Auth/Login/hooks-login/useRememberMe.ts:35]: ditto

File: src/components/pages/Auth/Login/hooks-login/useRememberMe.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  return data !== null ? JSON.parse(data || '') : { tag: '', password: '' };
```
Comment (ibraheemosule) [inline src/components/pages/Auth/Login/hooks-login/useRememberMe.ts:36]:
```suggestion
  if(data) return  JSON.parse(data)
```

File: src/components/pages/Auth/Login/hooks-login/useRememberMe.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+}: TRememberMeProps) => {
+  useEffect(() => {
+    if (rememberMe) {
+      saveUser({ tag, password });
+    }
+  }, [tag, password, rememberMe]);
+
+  useEffect(() => {
+    const user = isUserSaved();
+    const pword = CryptoJS.AES.decrypt(user.password, key);
```
Comment (ibraheemosule) [inline src/components/pages/Auth/Login/hooks-login/useRememberMe.ts:54]: create a reusable function in the utils to encrypt and decrypt
and move the cryptojs logic there

File: src/components/pages/Auth/Login/index.tsx [hunk 8]
```diff
@@ -6,14 +6,19 @@ import Auth from '..';
 import TextField from '~components/reusables/CustomField/TextField';
 import Api from '~api';
 import authStore from '~src/store/auth';
+import useRememberMe from '~components/pages/Auth/Login/hooks-login/useRememberMe';
 
 const { api } = new Api();
 
 const Login = () => {
   const { update } = authStore((state) => state);
+  console.log(update);
```
Comment (ibraheemosule) [inline src/components/pages/Auth/Login/index.tsx:15]: remove the console.log statement

**Review by ibraheemosule — CHANGES_REQUESTED** [review]

File: src/components/pages/Auth/Login/hooks-login/useRememberMe.ts (hunk 2 above)
Comment (Akintobi41) [reply src/components/pages/Auth/Login/hooks-login/useRememberMe.ts:20]: done

File: src/components/pages/Auth/Login/index.tsx (hunk 8 above)
Comment (Akintobi41) [reply src/components/pages/Auth/Login/index.tsx:15]: > remove the console.log statement

should i remove the update variable as well, it is not being used

File: src/components/pages/Auth/Login/hooks-login/useRememberMe.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      saveUser({ tag, password });
+    }
+  }, [tag, password, rememberMe]);
+
+  useEffect(() => {
+    const user = isUserSaved();
+
+    if (user?.tag) {
+      setTag(user.tag);
+      setPassword(decryptInput(user.password, key));
```
Comment (ibraheemosule) [inline src/components/pages/Auth/Login/hooks-login/useRememberMe.ts:57]: pass the setRememberMe here and set it to true
setRememberMe(true)

**Review by ibraheemosule — APPROVED** [review]

**Pull request merged**